lib/faraday/rack_builder.rb

```diff
@@ -84,6 +84,7 @@ def use(klass, *args, &block)
         use_symbol(Faraday::Middleware, klass, *args, &block)
       else
         raise_if_locked
+        raise_if_adapter_set unless is_adapter?(klass)
```

Comment:
Can you please explain this `unless is_adapter?(klass)`?
Does that mean that I'm allowed to insert an adapter after another adapter?